internal/release/services/release/release.go

```diff
@@ -258,13 +258,126 @@ func (s *ReleaseService) GetLatestRelease(ctx context.Context, request *releaseS
 		return releases[i].CreatedAt > releases[j].CreatedAt
 	})
 
-	// Return only the latest release
 	grpcResponse := releaseSvc.ListReleasesResponse{}
 	grpcResponse.Releases = append(grpcResponse.Releases, releases[0])
 
 	return &grpcResponse, nil
 }
 
+// GetDistinctValues is a helper function to filter the response and return only distinct values.
+func GetDistinctValues(resp []string) []string {
+	temp := make(map[string]bool)
+
+	for _, item := range resp {
+		temp[item] = true
+	}
+
+	var distinctList []string
+	for i := range temp {
+		distinctList = append(distinctList, i)
+	}
+
+	return distinctList
+}
+
+// ListReleaseTypes is used to retrieve all distinct release types.
+func (s *ReleaseService) ListReleaseTypes(ctx context.Context, request *releaseSvc.ListReleaseTypesRequest) (*releaseSvc.ListReleaseTypesResponse, error) {
+	span := trace.SpanFromContext(ctx)
+	span.SetAttributes(
+		attribute.String("release.page", request.GetPage().String()),
+	)
+
+	scanQueryInputs := &dynamodb.ScanInput{
+		ProjectionExpression: aws.String("#t"),
+		ExpressionAttributeNames: map[string]string{
+			"#t": "type",
+		},
+		TableName: aws.String(ReleaseTableName),
+	}
+
+	response, err := s.Db.Scan(ctx, scanQueryInputs)
+	if err != nil {
+		span.RecordError(err)
+		return nil, err
+	}
+
+	if response.Items == nil || len(response.Items) < 1 {
+		return nil, fmt.Errorf("nothing to retrieve '%v'", request.GetPage())
+	}
```

Comment:
We don't need this `if` as an empty response isn't an error and your later code will handle this condition.